run.py

```diff
@@ -156,6 +158,228 @@ def reindex(since=None, batch_size=None, force=False, update_solr=True, update_m
             logger.error('Failed while submitting data to pipeline')
         raise e
 
+def compare_solr(validation_logger, bibcodelist=None,filename=None):
+
+    if (bibcodelist is None) & (filename is None):
+        validation_logger.error('Must pass in a list of bibcodes or a file of bibcodes')
+
+    SOLR_OLD = app.conf['SOLR_URL_OLD']
+    SOLR_NEW = app.conf['SOLR_URL_NEW']
+
+    if bibcodelist is None:
+        bibcodelist = []
+
+    if filename is None:
+        lines = []
+    else:
+        with open(filename) as f:
+            lines = f.readlines()
+        lines = map(str.strip, lines)
+
+    bibcodes = bibcodelist + lines
+
+    t1 = time.time()
+
+    for bibcode in bibcodes:
+
+        r1 = query_solr(validation_logger, SOLR_OLD[0], 'bibcode:' + bibcode, sort="bibcode desc", fl='*')
```

Comment:
better to use quotes: 'bibcode:"' + bibcode + '"'

that's because some bibcodes contain '&'